contents/performance/PerformanceProfiling.md

```diff
@@ -0,0 +1,176 @@
+# Performance Profiling
+
+Author: [Ong Heng Le](https://github.com/initialshl)
+
+<br>
+
+# Overview
+
+Software profiling is collecting data about a program's execution, by sampling statistical 
+data during the program's execution or by modifying the program itself. Profilers may 
+collect data such as function timings, memory allocation and CPU usage in approximated 
+or exact details. 
+
+Performance profiling mostly involves sampling the CPU and collecting function timings 
+to identify portions in the program which takes up a significant amount of execution time. 
+These can be used to identify potential areas for code optimization.
+
+### Resources:
+1. [Profiling Overview](https://msdn.microsoft.com/en-us/library/bb384493(v=vs.110).aspx)
+
+<br>
+
+# Performance Profiling For The First Time
+
+Most profilers are similar in their functionalities, user interface, and use of 
+technical terms. You are encouraged to try this tutorial using your preferred profiling 
+tools on your own project, while making neccessary adjustments when required. <br>
```

Comment:
Can be more concise:
`You are encouraged to adapt this tutorial to your preferred profiling tools on your own project.`